mantle/kola/tests/misc/network.go

```diff
@@ -219,6 +229,81 @@ var (
           umount /var/mnt
 	`
 
+	setupVethPairsTemplate = `#!/usr/bin/env bash
+          set -ex
+
+          create_veth_pair() {
+            veth_host_side_end_name=$1
+            veth_netns_side_end_name=$2
+            host_side_mac_address=$3
+            netns_side_ip=$4
+            network_namespace=$5
+
+            # Create veth pair and assign a namespace to veth-netns
+            ip link add ${veth_host_side_end_name} type veth peer name ${veth_netns_side_end_name}
+            ip link set ${veth_netns_side_end_name} netns ${network_namespace}
+
+            # Assign an MAC address to the host-side veth end
+            ip link set dev ${veth_host_side_end_name} address ${host_side_mac_address}
+
+            # Assign an IP address to netns-side veth end and bring it up
+            ip netns exec ${network_namespace} ip address add ${netns_side_ip} dev ${veth_netns_side_end_name}
+            ip netns exec ${network_namespace} ip link set ${veth_netns_side_end_name} up
+          }
+
+          activate_veth_end() {
+            veth_end_name=$1
+
+            nmcli dev set ${veth_end_name} managed yes
+            ip link set ${veth_end_name} up
+
+            poll_dhcp_success ${veth_end_name}
+          }
+
+          poll_dhcp_success() {
+            veth_end_name=$1
+            for attempt in {1..5}; do
+              cidr=$(nmcli -g IP4.ADDRESS dev show ${veth_end_name})
+              if [[ ! -z ${cidr} ]]; then
+                return
+              fi
+              sleep 1
+            done
+
+            echo "failed to get ip for ${veth_end_name}"
+            exit 1
+          }
+
+          primary_mac=%s
+          secondary_mac=%s
+          primary_ip=%s
+          secondary_ip=%s
+
+          network_namespace=test-netns
+          # Create a network namespace
+          ip netns add ${network_namespace}
+
+          # create 2 veth pairs between host side and network_namespace
+          veth1_host_side_end_name=veth1-host
+          veth1_netns_side_end_name=veth1-netns
+          create_veth_pair ${veth1_host_side_end_name} ${veth1_netns_side_end_name} ${primary_mac} 192.168.0.100 ${network_namespace}
+          veth2_host_side_end_name=veth2-host
+          veth2_netns_side_end_name=veth2-netns
+          create_veth_pair ${veth2_host_side_end_name} ${veth2_netns_side_end_name} ${secondary_mac} 192.168.0.101 ${network_namespace}
+
+          # Run a dnsmasq service on the network_namespace, to set the host-side veth ends a ip via their MAC addresses
+          echo -e "dhcp-range=192.168.0.50,192.168.0.60,255.255.255.0,12h\ndhcp-host=${primary_mac},${primary_ip}\ndhcp-host=${secondary_mac},${secondary_ip}" > /etc/dnsmasq.d/dhcp
+          ip netns exec ${network_namespace} dnsmasq --log-queries --log-facility=/var/log/dnsmasq.log &
```

Comment:
How do we recover these logs?

Reply:
When I was debugging using this, I just added this
```
output := string(c.MustSSH(m, "cat /var/log/dnsmasq.log"))
c.Fatal(output)
```
it makes the test temporarily fail, but it's good enough for debugging imo

Reply:
Do we need to carry dead code there then? If you have to make changes to the code to use it, why would not you do what you describe above and add the `--log-facility`

Reply:
I agree :) Done